app/src/test/java/tech/relaycorp/courier/domain/client/CargoCollectionTest.kt

```diff
@@ -0,0 +1,50 @@
+package tech.relaycorp.courier.domain.client
+
+import com.nhaarman.mockitokotlin2.any
+import com.nhaarman.mockitokotlin2.eq
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import tech.relaycorp.courier.data.database.StoredMessageDao
+import tech.relaycorp.courier.data.disk.DiskRepository
+import tech.relaycorp.courier.data.model.MessageType
+import tech.relaycorp.courier.domain.DeleteMessage
+import tech.relaycorp.courier.domain.StoreMessage
+import tech.relaycorp.courier.test.factory.StoredMessageFactory
+import tech.relaycorp.relaynet.cogrpc.client.CogRPCClient
+
+internal class CargoCollectionTest {
```

Comment:
I'd suggest adding tests for the following cases:

- The nominal case. When the CCA is accepted and zero to many cargoes are returned.
- The scenario where the call fails due to a reason other than `PERMISSION_DENIED` -- For example, `UNAVAILABLE`.

Note that the test `collect with CCA refused deletes CCA` is passing because CCAs are always deleted in the current version of the code -- That's why I think the second test is particularly important. (As a reminder, CCAs should only be deleted if they're fulfilled by the server or the server refuses them as invalid with a `PERMISSION_DENIED`; otherwise, they should be retained so they can be retried on the next sync)